lib/middleware/redirectToExistingSession.js

```diff
@@ -3,24 +3,31 @@ const logger = require('../logger');
 
 module.exports = (req, res, next) => {
     const operation = 'redirectToExistingSession';
-    logger.info({operation, KATsessionExists: req.session.isPopulated, user: JSON.stringify(req.currentUser)});
+    logger.info({operation, KATsessionExists: req.session ? req.session.isPopulated: undefined, user: JSON.stringify(req.currentUser)});
 
-    // if the licence is not valid but we have the KMT session => redirect to the first licence
-    if (req.session && req.session.kmtLoggedIn && Array.isArray(req.listOfLicences) && req.listOfLicences[0] && req.listOfLicences[0].licenceId) {
-        logger.info({operation, licenceToRedirectTo: req.listOfLicences[0].licenceId});
+    return new Promise((resolve, reject) => {
+        if (req.session) {
+            // if the licence is not valid but we have the KMT session => redirect to the first licence
+            if (req.session.kmtLoggedIn && Array.isArray(req.listOfLicences) && req.listOfLicences[0] && req.listOfLicences[0].licenceId) {
```

Comment:
Could probably do with caching `req.listOfLicences` as we refer to it quite often.